usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/TestHttpCallImpl.java

```diff
@@ -66,31 +68,35 @@ public void start(Collection<? extends Location> locations) {
         }
     }
 
-    private Supplier<String> buildDataSupplier(final HttpAssertionTarget httpAssertionTarget, final String url) {
-
-        switch (httpAssertionTarget) {
+    private void doRequestAndCheckAssertions(Map<String, Duration> flags, List<Map<String, Object>> assertions,
+                                             HttpAssertionTarget target, final String url) {
+        switch (target) {
             case body:
-                return new Supplier<String>() {
+                Supplier<String> getBody = new Supplier<String>() {
                     @Override
                     public String get() {
                         return HttpTool.getContent(url);
                     }
                 };
+                TestFrameworkAssertions.checkAssertions(flags, assertions, target.toString(), getBody);
+                break;
             case status:
-                return new Supplier<String>() {
+                Supplier<Integer> getStatusCode = new Supplier<Integer>() {
                     @Override
-                    public String get() {
+                    public Integer get() {
                         try {
-                            return String.valueOf(HttpTool.getHttpStatusCode(url));
+                            return HttpTool.getHttpStatusCode(url);
                         } catch (Exception e) {
-                            LOG.error("HTTP call to [{}] failed due to [{}] ... returning Status code [0 - Unreachable]", url, e.getMessage());
-                            return "0";
+                            LOG.error("HTTP call to [{}] failed due to [{}] ... returning Status code [500]",
```

Comment:
definitely not ERROR level, since here we're just in a status check.

normally in catch blocks where we swallow generic exceptions we include

        Exceptions.propagateIfFatal(e);

in case it's a severe exception.  but here' i'd simply propagate it regardless (`Exceptions.propagate(e)') and make sure the wrapper code treats errors appropriately.  returning `500` is dishonest :) .